### PR TITLE
SWC-134

### PR DESCRIPTION
**Pull request by KadenZipfel**
- Same CWE used as SWC-116; perhaps there's a better one.

### CODE REVIEW

File: entries/SWC-134.md [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+Avoid Using Block Number as a Proxy for Time
+
+## Relationships
+[CWE-829: Inclusion of Functionality from Untrusted Control Sphere](https://cwe.mitre.org/data/definitions/829.html)
+
+## Description
+Generally, the block time on Ethereum is about 14 seconds. Given this knowledge it's possible to predict the time delta between blocks. However, block times are not constant and are subject to change for a variety of reasons, e.g. fork reorganisations and the difficulty bomb. Due to variable block times, `block.number` should not be relied on for precise calculations of time. 
+
+## Remediation
+Precise calculations of time should not be done using `block.number`. Instead, developers may make use of oracles.
```
Comment (thec00n) [inline entries/SWC-134.md]: I think using `block.timestamp` is acceptable for some use cases and use of an oracle whenever you deal with some sort time period period might be overkill?

**Comment by thec00n** [comment]
Also it seems that the proposed category seems to be included in https://swcregistry.io/docs/SWC-116. Why do you think we need another category?

**Comment by KadenZipfel** [comment]
> Also it seems that the proposed category seems to be included in https://swcregistry.io/docs/SWC-116. Why do you think we need another category?

Fair point. Perhaps it would be best to combine them, changing the title to something along the lines of, "Avoid block values as a proxy for time".

**Comment by maurelian** [comment]
That's also consistent with the way the best practices addresses the various issues with timestamps [here](https://consensys.github.io/smart-contract-best-practices/recommendations/#timestamp-dependence)

**Comment by thec00n** [comment]
`Fair point. Perhaps it would be best to combine them, changing the title to something along the lines of, "Avoid block values as a proxy for time".`
Generally the weakness titles of the SWC categories refer to the issue not the remediation.

**Comment by KadenZipfel** [comment]
So I think the best path forward would be to deprecate SWC-116 as per https://github.com/SmartContractSecurity/SWC-registry/issues/234 then create a new SWC titled, "Block values as a proxy for time".

**Comment by maurelian** [comment]
Is that better than renaming SWC-116 to "Block values as a proxy for time", and perhaps modify the text a slightly to make it more general to both block.number and block.timestamp?

**Comment by KadenZipfel** [comment]
> Is that better than renaming SWC-116 to "Block values as a proxy for time", and perhaps modify the text a slightly to make it more general to both block.number and block.timestamp?

I can't believe I hadn't thought of this. Will do that and scrap this PR.

**Pull request closed without merging**